kasa/smartdevice.py

```diff
@@ -382,6 +471,9 @@ def update(d, u):
     @requires_update
     def emeter_today(self) -> Optional[float]:
         """Return today's energy consumption in kWh."""
+        if not self.has_emeter:
+            raise SmartDeviceException("Device has no emeter")
```

Comment:
I know this WIP, but shouldn't this be in a separate PR?

Reply:
I think it's such a minor change (just to avoid crashing if someone tries to access those), that it doesn't really matter.

Reply:
Ah, so this is related, and not a separate bugfix. Makes sense then!

Reply:
So, there are a few minor things I noticed thanks to having those doctests (like this one, and the invalid return value for SmartStripPlug's `is_on`), that I think were worth fixing here rather than separately.